crates/proof-of-sql/src/base/database/column_repetition_operation.rs

```diff
@@ -120,19 +136,48 @@ impl RepetitionOp for ColumnRepeatOp {
     fn op<T: Clone>(column: &[T], n: usize) -> impl Iterator<Item = T> {
         repeat_slice(column, n)
     }
+
+    fn op_fixed_size_binary(col_bytes: &[u8], width: usize, n: usize) -> Vec<u8> {
+        let num_rows = col_bytes.len() / width;
+        let mut out = Vec::with_capacity(col_bytes.len() * n);
+
+        for _ in 0..n {
+            for row_idx in 0..num_rows {
+                let start = row_idx * width;
+                let end = start + width;
+                out.extend_from_slice(&col_bytes[start..end]);
+            }
+        }
+        out
+    }
```

Comment:
You can shrink this down a bit with ```cycle```:
```suggestion
fn op_fixed_size_binary(col_bytes: &[u8], width: usize, n: usize) -> Vec<u8> {
    let rows: Vec<_> = col_bytes.chunks_exact(width).collect();
    let mut out = Vec::with_capacity(rows.len() * width * n);
    for row in rows.iter().cycle().take(rows.len() * n) {
        out.extend_from_slice(row);
    }
    out
}
```